pkg/odo/genericclioptions/runnable.go

```diff
@@ -30,15 +30,29 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 
 // CheckMachineReadableOutputCommand performs machine-readable output functions required to
 // have it work correctly
-func CheckMachineReadableOutputCommand() {
+func CheckMachineReadableOutputCommand(cmd *cobra.Command) {
 
 	// Check that the -o flag has been correctly set as json.
 	outputFlag := pflag.Lookup("o")
-	if outputFlag != nil && outputFlag.Changed && outputFlag.Value.String() != "json" {
+	flagChanged := outputFlag != nil && outputFlag.Changed
+	machineOutput := cmd.Annotations["machineoutput"]
+
+	if flagChanged && outputFlag.Value.String() != "json" {
 		log.Error("Please input a valid output format for -o, available format: json")
```

Comment:
All these (log.Error + exit 1) could possibly be replaced by a [log.Fatal like the standard log package](https://golang.org/pkg/log/#Fatal) maybe. Anycase more of an implementation detail :)

Reply:
Yeah, we use `log.Error` instead since we're able to modify the output to something more pretty :)